.github/workflows/automake.yaml

```diff
@@ -0,0 +1,14 @@
+name: Tests and Build
+'on':
+  push:
+    branches:
+      - "*"
+    tags:
+      - v*
+jobs:
+  runs on: ubuntu-latest
+  steps:
+    - name: Checkout Code
+      uses: actions/checkout@v3
+    - name: Run Unit Test
```

Comment:
Do you need a step here like:

```
- name: Build
  run: make
```

sorry if I didn't mention that before... you probably should do `make` before just to make sure it all builds